components/post_view/post_list/index.jsx

```diff
@@ -0,0 +1,448 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {FormattedMessage} from 'react-intl';
+
+import LoadingScreen from 'components/loading_screen.jsx';
+import CreateChannelIntroMessage from 'components/post_view/channel_intro_message';
+import Constants from 'utils/constants.jsx';
+import {getNewestPostIdFromPosts, getOldestPostIdFromPosts} from 'utils/post_utils';
+
+import PostList from './post_list';
+
+const POSTS_PER_PAGE = Constants.POST_CHUNK_SIZE / 2;
+
+export default class PostListWrapper extends React.PureComponent {
+    static propTypes = {
+
+        /**
+         * Array of posts in the channel, ordered from newest to oldest
+         */
+        posts: PropTypes.array,
```

Comment:
Shouldn't this be required?

Reply:
No. Posts can be `null` in the case for first channel loading and this component is responsible for loading them so required did not seem al that right here.